clients/dataset/dataset.go

```diff
@@ -88,6 +86,11 @@ func (c *Client) Get(ctx context.Context, id string) (m Model, err error) {
 
 	req = setCollectionID(ctx, req)
 
+	if len(authToken) > 0 {
```

Comment:
I would consider moving this out into its own function which can be reused OR move the len check and the logging into `common.AddServiceTokenHeader`. That was the overhead is all in that function and you can call it where ever

Reply:
simplified as per slack chat.